handler/src/main/java/io/netty/handler/ssl/util/BouncyCastlePemReader.java

```diff
@@ -0,0 +1,169 @@
+ * under the License.
+ */
+
+package io.netty.handler.ssl.util;
+
+import io.netty.util.CharsetUtil;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.AccessController;
+import java.security.PrivateKey;
+import java.security.PrivilegedAction;
+import java.security.Provider;
+import java.security.Security;
+
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
+import org.bouncycastle.openssl.PEMDecryptorProvider;
+import org.bouncycastle.openssl.PEMEncryptedKeyPair;
+import org.bouncycastle.openssl.PEMException;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.bouncycastle.openssl.jcajce.JceOpenSSLPKCS8DecryptorProviderBuilder;
+import org.bouncycastle.openssl.jcajce.JcePEMDecryptorProviderBuilder;
+import org.bouncycastle.operator.InputDecryptorProvider;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.pkcs.PKCS8EncryptedPrivateKeyInfo;
+import org.bouncycastle.pkcs.PKCSException;
+
+public final class BouncyCastlePemReader {
+
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(BouncyCastlePemReader.class);
+    private static Boolean available;
+
+    public static boolean isBcAvailable() {
+        tryLoadingBc();
+        return available;
+    }
+
+    private static void tryLoadingBc() {
+        if (available != null) {
+            return;
+        }
+
+        available = AccessController.doPrivileged(new PrivilegedAction<Boolean>() {
+            @Override
+            public Boolean run() {
+                try {
+                    Class<Provider> bcProviderClass
+                      = (Class<Provider>) Class.forName("org.bouncycastle.jce.provider.BouncyCastleProvider"
+                              , true, this.getClass().getClassLoader());
+                    if (Security.getProvider("BC") == null) {
+                        Security.addProvider(bcProviderClass.newInstance());
+                    }
+                    logger.debug("Bouncy Castle provider available");
+                    return Boolean.TRUE;
+                } catch (Exception e) {
+                    logger.debug("Cannot load Bouncy Castle provider");
+                    return Boolean.FALSE;
+                }
+            }
+        });
+    }
+
+    static {
+        isBcAvailable();
+    }
+
+    public static PrivateKey getPrivateKey(InputStream keyInputStream, String keyPassword) {
+        try {
+            return getPrivateKey(getParser(keyInputStream), keyPassword);
+        } catch (Exception e) {
+            logger.debug("Unable to extract private key due to " + e);
```

Comment:
change to: `logger.debug("Unable to extract private key", e);`

This way we also get a stacktrace